CMakeLists.txt

```diff
@@ -368,6 +378,10 @@ if (NOT BUILD_SHARED_LIBS)
 endif ()
 if (BUILD_SHARED_LIBS)
   set (HDF5_VOL_REST_LIBRARIES_TO_EXPORT ${HDF5_VOL_REST_LIBRARIES_TO_EXPORT} ${HDF5_VOL_REST_LIBSH_TARGET} CACHE INTERNAL "Store which libraries should be exported" FORCE)
+  # Value is not automatically inherited from cache on Windows
```

Comment:
This and the same below must either be a CMake bug or something wrong in the usage of this variable in the CMake code somewhere. CMake behavior doesn't generally change across platforms, except for things like quoting and other edge cases. The set call above should be enough for any case needed, though it is a bit heavy-handed.